Drive/rover_drive_system.hpp

```diff
@@ -150,19 +150,104 @@ class RoverDriveSystem
+       {
+        auto next_same_time = Uptime() + 100ms;
+
+        //current_speed is pulled from the motor feedback and casted to long double 
+        currentSpeed_leftWheel = static_cast<long double>(left_wheel_.RequestFeedbackFromMotor().GetFeedback().speed);
+        currentSpeed_rightWheel = static_cast<long double>(right_wheel_.RequestFeedbackFromMotor().GetFeedback().speed);
+        currentSpeed_backWheel = static_cast<long double>(back_wheel_.RequestFeedbackFromMotor().GetFeedback().speed);
+        
+        //lerp returns a midpoint between current_speed and the new_speed
+        lerpSpeed_leftWheel = std::lerp(currentSpeed_leftWheel, new_speed, static_cast<long double>(0.5));
+        lerpSpeed_rightWheel = std::lerp(currentSpeed_rightWheel, new_speed, static_cast<long double>(0.5));
+        lerpSpeed_backWheel = std::lerp(currentSpeed_backWheel, new_speed, static_cast<long double>(0.5));
+
+        
+        //Set the speed of the motor and convert long double lerp_speed to revolutions_per_minute_t object
+        //use std::clamp to make sure we don't set the speed higher/lower than the input speed.
+        //When using std::clamp check if the fucntion is increasing or decreasing the speed to limit the max/min speeds
+        if(increaseSpeed_leftWheel)
+        {
+            left_wheel_.SetSpeed(std::clamp(units::angular_velocity::revolutions_per_minute_t{lerpSpeed_leftWheel}, kMaxNegSpeed, speed));
+        }
+        else
+        {
+            left_wheel_.SetSpeed(std::clamp(units::angular_velocity::revolutions_per_minute_t{lerpSpeed_leftWheel}, speed, kMaxPosSpeed));
+        }
+        if(increaseSpeed_rightWheel)
+        {
+            right_wheel_.SetSpeed(std::clamp(units::angular_velocity::revolutions_per_minute_t{lerpSpeed_rightWheel}, kMaxNegSpeed, speed));
+        }
+        else
+        {
+            right_wheel_.SetSpeed(std::clamp(units::angular_velocity::revolutions_per_minute_t{lerpSpeed_rightWheel}, speed, kMaxPosSpeed));
+        }
+        if(increaseSpeed_backWheel)
+        {
+            back_wheel_.SetSpeed(std::clamp(units::angular_velocity::revolutions_per_minute_t{lerpSpeed_backWheel}, kMaxNegSpeed, speed));
+        }
+        else
+        {
+            back_wheel_.SetSpeed(std::clamp(units::angular_velocity::revolutions_per_minute_t{lerpSpeed_backWheel}, speed, kMaxPosSpeed));
+        }
+        
+        
+
+
+        sjsu::LogInfo("Set wheel speed to %f for all wheels", lerp_speed);
+
+        //make sure 100ms have passed before setting the next speed
+        while(Uptime() < next_same_time)
+        {
+          continue;
+        }
```

Comment:
Yeah this won't work with the rest of the code as the system is locked here waiting for time to pass.